providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/sync/SyncStreamQueueSource.java

```diff
@@ -115,82 +111,57 @@ public void shutdown() throws InterruptedException {
-
-                restart(); // start the stream with the context
 
-                // TODO: remove the metadata call entirely after https://github.com/open-feature/flagd/issues/1584
-                if (!syncMetadataDisabled) {
-                    try {
-                        FlagSyncServiceBlockingStub localStub = blockingStub;
-
-                        if (deadline > 0) {
-                            localStub = localStub.withDeadlineAfter(deadline, TimeUnit.MILLISECONDS);
-                        }
-
-                        metadataResponse = localStub.getMetadata(metadataRequest.build());
-                    } catch (Exception metaEx) {
-                        // cancel the stream if the getMetadata fails
-                        // we can keep this log quiet since the stream is cancelled/restarted with this exception
-                        log.debug("Metadata exception: {}, cancelling stream", metaEx.getMessage(), metaEx);
-                        context.cancel(metaEx);
+            try (Context.CancellableContext context = Context.current().withCancellation()) {
+                SyncStreamObserver observer = new SyncStreamObserver(outgoingQueue, context);
+                Context previous = context.attach();
+                try {
+                    if (!syncMetadataDisabled) {
+                        observer.metadata = getMetadata(context);
                     }
+
+                    syncFlags(observer);
+                } finally {
+                    context.detach(previous);
                 }
+            }
+        }
 
-                // inner loop for handling messages
-                while (!shutdown.get() && !Context.current().isCancelled()) {
-                    final StreamResponseModel<SyncFlagsResponse> taken = incomingQueue.take();
-                    if (taken.isComplete()) {
-                        log.debug("Sync stream completed, will restart");
-                        // The stream is complete, we still try to reconnect
-                        break;
-                    }
+        log.info("Shutdown invoked, exiting event stream listener");
+    }
 
-                    Throwable streamException = taken.getError();
-                    if (streamException != null) {
-                        log.debug("Exception in stream RPC, streamException {}, will restart", streamException);
-                        if (!outgoingQueue.offer(new QueuePayload(
-                                QueuePayloadType.ERROR,
-                                String.format("Error from stream: %s", streamException.getMessage())))) {
-                            log.error("Failed to convey ERROR status, queue is full");
-                        }
-                        break;
-                    }
+    // TODO: remove the metadata call entirely after https://github.com/open-feature/flagd/issues/1584
+    private static final GetMetadataRequest.Builder metadataRequest = GetMetadataRequest.newBuilder();
```

Comment:
```suggestion
    private static final GetMetadataRequest.Builder metadataRequestBuilder = GetMetadataRequest.newBuilder();
```
This should be at the very top of the file

Reply:
I got rid of the whole builder by using `GetMetadataRequest.getDefaultInstance()`.